integration-tests/utils/globals.js

```diff
@@ -86,11 +86,12 @@ module.exports = {
 
     Object.assign(browser.globals, await buildBackendContext(browser));
 
-    browser.globals.emulateTabSwitchingAway = async () => {
-      await browser.execute(() => {
-        Object.defineProperty(document, 'visibilityState', { value: 'hidden', configurable: true });
-        document.dispatchEvent(new Event('visibilitychange'));
-      });
+    browser.globals.emulateTabSwitching = async (visible) => {
+      await browser.execute((hidden) => {
+        Object.defineProperty(document, 'hidden', { value: hidden, configurable: true });
```

Comment:
Looking at this, I'd suggest we make it a unit test. This isn't how an integration test should work.

Reply:
Will look into making it a real integration test with a real tab switch at some point